openwisp_notifications/tests/test_selenium.py

```diff
@@ -126,45 +126,71 @@ def test_email_unsubscribe_page(self):
-            By.CSS_SELECTOR,
-            '.global-setting-dropdown[data-web-state] .global-setting-dropdown-toggle',
-        ).click()
-        self.find_element(
-            By.CSS_SELECTOR,
-            '.global-setting-dropdown[data-web-state]'
-            ' .global-setting-dropdown-menu button:last-child',
-        ).click()
-        self.find_element(By.CSS_SELECTOR, '#confirmation-modal #confirm').click()
-
-        # Expand the first organization panel if it's collapsed
-        first_org_toggle = self.find_element(By.CSS_SELECTOR, '.toggle-icon')
-        first_org_toggle.click()
-
-        all_checkboxes = self.find_elements(
-            By.CSS_SELECTOR, 'input[type="checkbox"]', wait_for='presence'
-        )
-        for checkbox in all_checkboxes:
-            self.assertFalse(checkbox.is_selected())
 
-        # Check the org-level web checkbox
-        org_level_web_checkbox = self.find_element(By.CSS_SELECTOR, '#org-1-web')
-        org_level_web_checkbox.click()
+        with self.subTest('Notifications are enabled'):
+            # Expand the first organization panel
+            self.find_element(By.CSS_SELECTOR, '.toggle-icon').click()
+            # All web notifications are enabled
+            web_checkboxes = self.find_elements(
+                By.CSS_SELECTOR,
+                'input[type="checkbox"][data-column="web"]',
+                wait_for='presence',
+            )
+            for checkbox in web_checkboxes:
+                self.assertEqual(checkbox.is_selected(), True)
+            # Email notification is enabled for default type
+            default_email_checkbox = self.find_element(
+                By.CSS_SELECTOR,
+                'label#org-1-email-1 input',
+                wait_for='presence',
+            )
+            self.assertTrue(default_email_checkbox.is_selected())
+            # Email notification is disabled for generic type
+            generic_email_checkbox = self.find_element(
+                By.CSS_SELECTOR,
+                'label#org-1-email-2 input',
+                wait_for='presence',
+            )
+            self.assertFalse(generic_email_checkbox.is_selected())
+
+        with self.subTest('Disabling global notification setting'):
+            self.find_element(
+                By.CSS_SELECTOR,
+                '.global-setting-dropdown[data-web-state] .global-setting-dropdown-toggle',
+            ).click()
+            self.find_element(
+                By.CSS_SELECTOR,
+                '.global-setting-dropdown[data-web-state]'
+                ' .global-setting-dropdown-menu button:last-child',
+            ).click()
+            self.find_element(By.CSS_SELECTOR, '#confirmation-modal #confirm').click()
+
+            all_checkboxes = self.find_elements(
+                By.CSS_SELECTOR, 'input[type="checkbox"]', wait_for='presence'
+            )
+            for checkbox in all_checkboxes:
+                self.assertFalse(checkbox.is_selected())
```

Comment:
Can we take advantage to ensure the values in the database matches what we expect?